definitions/output/all/reprocess_pages.js

```diff
@@ -0,0 +1,262 @@
+);
+`);
+
+
+const iterations = [];
+const clients = constants.clients;
+
+for (
+  let month = constants.current_month;
+  month >= '2024-09-01'; // 2022-07-01
+  month = constants.fn_past_month(month)) {
+    clients.forEach((client) => {
+      iterations.push({
+        month: month,
+        client: client
+        })
+    })
+}
+
+iterations.forEach((iteration, i) => {
+  operate(`all_pages_stable_update ${iteration.month} ${iteration.client}`).tags([
+    "all_pages_stable"
+  ]).dependencies([
+    i===0 ? "all_pages_stable_pre" : `all_pages_stable_update ${iterations[i-1].month} ${iterations[i-1].client}`
+  ]).queries(ctx => `
+INSERT INTO \`all_dev.pages_stable\`
+SELECT
+  date,
+  client,
+  page,
+  is_root_page,
+  root_page,
+  rank,
+  wptid,
+  JSON_REMOVE(
+    SAFE.PARSE_JSON(payload, wide_number_mode => 'round'),
+    '$._metadata',
+    '$._detected',
+    '$._detected_apps',
+    '$._detected_technologies',
+    '$._detected_raw',
+    '$._custom',
+    '$._00_reset',
+    '$._a11y',
+    '$._ads',
+    '$._almanac',
+    '$._aurora',
+    '$._avg_dom_depth',
+    '$._cms',
+    '$._Colordepth',
+    '$._cookies',
+    '$._crawl_links',
+    '$._css-variables',
+    '$._css',
+    '$._doctype',
+    '$._document_height',
+    '$._document_width',
+    '$._Dpi',
+    '$._ecommerce',
+    '$._element_count',
+    '$._event-names',
+    '$._fugu-apis',
+    '$._generated-content',
+    '$._has_shadow_root',
+    '$._Images',
+    '$._img-loading-attr',
+    '$._initiators',
+    '$._inline_style_bytes',
+    '$._javascript',
+    '$._lib-detector-version',
+    '$._localstorage_size',
+    '$._markup',
+    '$._media',
+    '$._meta_viewport',
+    '$._num_iframes',
+    '$._num_scripts_async',
+    '$._num_scripts_sync',
+    '$._num_scripts',
+    '$._observers',
+    '$._origin-trials',
+    '$._parsed_css',
+    '$._performance',
+    '$._privacy-sandbox',
+    '$._privacy',
+    '$._pwa',
+    '$._quirks_mode',
+    '$._Resolution',
+    '$._responsive_images',
+    '$._robots_meta',
+    '$._robots_txt',
+    '$._sass',
+    '$._security',
+    '$._sessionstorage_size',
+    '$._structured-data',
+    '$._third-parties',
+    '$._usertiming',
+    '$._valid-head',
+    '$._well-known',
+    '$._wpt_bodies',
+    '$._blinkFeatureFirstUsed'
+  ) AS payload,
+  JSON_REMOVE(
+    SAFE.PARSE_JSON(summary, wide_number_mode => 'round'), 
+    '$._adult_site',
+    '$.archive',
+    '$.avg_dom_depth',
+    '$.crawlid',
+    '$.createDate',
+    '$.doctype',
+    '$.document_height',
+    '$.document_width',
+    '$.label',
+    '$.localstorage_size',
+    '$.meta_viewport',
+    '$.metadata',
+    '$.num_iframes',
+    '$.num_scripts_async',
+    '$.num_scripts_sync',
+    '$.num_scripts',
+    '$.pageid',
+    '$.PageSpeed',
+    '$.rank',
+    '$.sessionstorage_size',
+    '$.startedDateTime',
+    '$.url',
+    '$.urlhash',
+    '$.urlShort',
+    '$.usertiming',
+    '$.wptid',
+    '$.wptrun'
```

Comment:
Some of these DO still look to be used. Not sure what I was thinking in https://github.com/HTTPArchive/data-pipeline/issues/149#issuecomment-1317208562 

```suggestion
    '$._adult_site',
    '$.archive',
    '$.label',
    '$.metadata',
    '$.pageid',
    '$.PageSpeed',
    '$.rank',
    '$.url',
    '$.urlhash',
    '$.urlShort',
    '$.wptid',
```

It's arguable they belong more in custom_metrics but since they are gathered by WebPageTest itself rather than one of our own custom metrics they should stay here.

Reply:
I don't see them among [WPT](https://docs.webpagetest.org/metrics/page-metrics/) metrics, but we create them on the crawler ([1](https://github.com/HTTPArchive/wptagent/blob/10afe496c1e06909888b12ed1d05648cbc138965/HTTPArchive/transformation.py#L578), [2](https://github.com/HTTPArchive/wptagent/blob/10afe496c1e06909888b12ed1d05648cbc138965/HTTPArchive/transformation.py#L379)).

And as the scenario here was to deduplicate metrics, maybe we could update the queries, wherever they were used? Especially those from the top level (`url`, `wptid`, `metadata`, `rank`) - wouldn't it be more efficient?

Reply:
So we still need to keep `crawlid, createDate, startedDateTime, wptrun`?

Reply:
I couldn't see them exposed anywhere else. Arguably they are more metadata than summary though so I'd be happy to move them there. It's also weird that the parent wptid is in meta data but the actual one is not! Which is another reason for moving these to there.